chain/client/src/client_actor.rs

```diff
@@ -1161,9 +1161,17 @@ impl ClientActorInner {
     pub(crate) fn check_triggers(&mut self, ctx: &mut dyn DelayedActionRunner<Self>) -> Duration {
         let _span = tracing::debug_span!(target: "client", "check_triggers").entered();
         if let Some(config_updater) = &mut self.config_updater {
-            config_updater.try_update(&|updateable_client_config| {
-                self.client.update_client_config(updateable_client_config)
-            });
+            config_updater.try_update(
+                &|updateable_client_config| {
+                    self.client.update_client_config(updateable_client_config)
+                },
+                &|validator_signer| self.client.update_validator_signer(validator_signer),
+            );
+            if config_updater.was_validator_signer_updated() {
```

Comment:
nit: Instead of this can you return a value from the `update_validator_signer`?

Reply:
Feel free to refactor it if it would make the code nicer e.g. instead of passing the lambda's just return whatever is needed and call client.update_* methods from here.

Reply:
`update_validator_signer` already returns a value but it is passed as a lambda to `try_update()`.
`try_update()` has a loop that reads from stream and can handle two kinds of updates. I could return two vectors from `try_update()` and write the loop here instead but IMO it would make the code less nice.

Reply:
Can try_update return a struct that indicates the parts of the config (client config, validator signer) that were updates, eg. a list of bools? That would eliminate keeping the bool internally as a field, since it is consumed right after try_update returns.